bolt/lib/Profile/BoltAddressTranslation.cpp

```diff
@@ -197,8 +197,10 @@ void BoltAddressTranslation::writeMaps(std::map<uint64_t, MapTy> &Maps,
             ? SecondaryEntryPointsMap[Address].size()
             : 0;
     if (Cold) {
-      size_t HotIndex =
-          std::distance(ColdPartSource.begin(), ColdPartSource.find(Address));
+      // `Maps` is keyed by output addresses.
+      auto HotEntryIt = Maps.find(ColdPartSource[Address]);
+      assert(HotEntryIt != Maps.end());
```

Comment:
Does this happen when BAT maps are incomplete? Since BAT is part of the input, should this be an error instead?

Reply:
This is BAT writing code, so it's an internal invariant (should be an assertion).